aidermacs-models.el

```diff
@@ -22,123 +22,119 @@ Also based on aidermacs LLM benchmark: https://aidermacs.chat/docs/leaderboards/
+                           (alist-get 'models json-data)
+                         (alist-get 'data json-data))))
+          (mapcar (lambda (model)
+                    (concat prefix "/"
+                            (cond
+                             ((string= hostname "generativelanguage.googleapis.com")
+                              (replace-regexp-in-string "^models/" "" (alist-get 'name model)))
+                             ((stringp model) model)  ; Handle case where model is just a string
+                             (t (or (alist-get 'id model)
+                                    (alist-get 'name model))))))
+                  models))))))
 
-(defun aidermacs--get-supported-models ()
-  "Get list of models supported by aider using the /models command."
-  (let ((current-output aidermacs--current-output))
-    (aidermacs--send-command-backend (get-buffer (aidermacs-buffer-name)) "/models /")
-    ;; Wait briefly for output
-    (sleep-for 0.5)
-    (let* ((output aidermacs--current-output)
-           (models (seq-filter
-                   (lambda (line)
-                     (string-prefix-p "- " line))
-                   (split-string output "\n" t))))
-      (setq models (mapcar (lambda (line)
-                            (substring line 2)) ; Remove "- " prefix
-                          models))
-      (message "Supported models: %S" models)
-      models)))
 
-(defun aidermacs--get-available-models ()
-  "Get list of available models from multiple providers, using cache if available."
-  (when (and aidermacs--cached-models
-             (equal aidermacs--cached-models aidermacs-popular-models)
-             (fboundp 'aidermacs-buffer-name)
-             (get-buffer (aidermacs-buffer-name)))
-    (setq aidermacs--cached-models nil))
+(defun aidermacs--select-model ()
+  "Private function for model selection with completion."
+  (let ((model (with-local-quit
+                 (completing-read "Select AI model: " aidermacs--cached-models nil t))))
+    (when model
+      (aidermacs--send-command (format "/model %s" model) t))))
 
-  (unless aidermacs--cached-models
-    (if (and (fboundp 'aidermacs-buffer-name)
-             (get-buffer (aidermacs-buffer-name)))
-        (let ((models nil)
-              (supported-models (aidermacs--get-supported-models)))
-          (dolist (url '("https://api.openai.com/v1"
-                        "https://openrouter.ai/api/v1"
-                        "https://api.deepseek.com"
-                        "https://api.anthropic.com/v1"
-                        "https://generativelanguage.googleapis.com/v1beta"))
-            (condition-case err
-                (let* ((fetched-models (fetch-openai-compatible-models url))
-                       (filtered-models (seq-filter (lambda (model)
+(defun aidermacs--get-available-models ()
+  "Get list of models supported by aider using the /models command."
+  (aidermacs--send-command
+   "/models /" t
+   (lambda (output)
```

Comment:
FYI there is a `lambda` function to catch the output of the send-command, instead of utilizing `aidermacs--current-output`purely, because of vterm.